secret/scanner.go

```diff
@@ -72,6 +134,38 @@ func (s Scanner) Scan(args ScanArgs) types.Secret {
 	}
 }
 
+func shouldSkipPass(filePath string, allowPaths []*regexp.Regexp) bool {
+	for _, path := range allowPaths {
+		if path.MatchString(filePath) {
+			return true
+		}
+	}
+	return false
+}
+
+func removeAllowedSecrets(allRuleResults []RuleResult, allowedLocations [][]int) []RuleResult {
+	if len(allowedLocations) == 0 {
+		return allRuleResults
+	}
+
+	results := make([]RuleResult, 0)
+	for _, result := range allRuleResults {
+		if !isResultAllowed(result, allowedLocations) {
+			results = append(results, result)
+		}
+	}
+	return results
+}
+
+func isResultAllowed(result RuleResult, allowedLocations [][]int) bool {
+	for _, allowedLocation := range allowedLocations {
+		if result.StartPosition >= allowedLocation[0] && result.EndPosition <= allowedLocation[1] || result.StartPosition <= allowedLocation[0] && result.EndPosition >= allowedLocation[1] {
+			return true
```

Comment:
I realized we need to check it both ways.
Gitleaks searches for the allowed patterns only inside the match, for example:
found secret - `password: "defaultPassword"`, and the allow list has the regex `default`, it will ignore it.